addons/mail/controllers/mailbox.py

```diff
@@ -10,7 +10,7 @@ class MailboxController(http.Controller):
     def discuss_inbox_messages(self, search_term=None, before=None, after=None, limit=30, around=None):
         domain = [("needaction", "=", True)]
         res = request.env["mail.message"]._message_fetch(domain, search_term=search_term, before=before, after=after, around=around, limit=limit)
-        messages = res.pop("messages")
+        messages = res.pop("messages").filtered(lambda m: request.env[m.model].browse(m.res_id).exists())
```

Comment:
Are you going to make a query / record ? Inbox cross model / lot of IDs is going to be quite heavy ^^ . Group by model and do an exist / model (probably worth a small tool for that as it is going to be used in several controllers).